### PR TITLE
Upgrade Next.js and related packages

### PR DESCRIPTION
**Pull request by kachkaev**
## 🌟 What is the purpose of this PR?

The motivation for this PR is to replace webpack 4 with webpack 5 in HASH frontend (Next.js app). This unblocks @Alfred-Mountfield in https://github.com/hashintel/hash/pull/316. We can’t fix CI there because using webpack 5 in a new package causes some odd glitch between webpack 4 and webpack 5 and breaks Next.js.

This PR gives us a new version of Next.js, upgrades some Next-related dependencies and provides a slightly cleaner frontend setup.

## ⚠️ Known issues

- We now have webpack4 in blocks and webpack5 in Next.js. This would be fine if we did not `require("html-webpack-plugin")` in `next.config.js` for building block sanbox. Now that we have imports of `html-webpack-plugin` 4 and 5, `check-dependency-version-consistency` [complains](https://github.com/hashintel/hash/runs/5223019770?check_suite_focus=true).  I used `"html-webpack-plugin-v4": "npm:html-webpack-plugin@4.5.1"` in block `package.json` files, which fixes this issue without adding any special ignore rules. We will hopefully update block templates soon, so the problem will self-resolve. We need to do that anyway to get rid of weback 4.

   I will need to update this PR or https://github.com/hashintel/hash/pull/308 depending on which one gets merged. CI won’t let me forget 😅

## 🐾 Next steps

- Upgrade webpack to 5 in blocks – [Asana task](https://app.asana.com/0/0/1201836863408318/f) _(internal)_

- Sense-check Next.js plugins in Þ repo  – [Asana task](https://app.asana.com/0/0/1201815866759884/f) _(internal)_
  (related to a recent Next.js bump there - https://github.com/blockprotocol/blockprotocol/pull/216)

## 🔍 What does this change?

- Next 12.1 🚀

- swc instead of babel for faster dev experience
  - I removed `packages/hash/frontend/.babelrc`
  - I updated `packages/hash/frontend/jest.config.js` to use `const nextJest = require("next/jest")` (cool stuff!)

- `@svgr/webpack` instead of `next-images`
   - inspired by https://dev.to/dolearning/importing-svgs-to-next-js-nna#svgr
   - `<Svg />` is not `any` any more
   - no type clashes for static PNG imports etc (which are perfectly handled by Next itself)

- no more `FixMeLater` type (alias to `any`) – was used in two cases

- no `prismjs` dependency in frontend app – syntax is still highlighted:
  Screenshot 2022-02-16 at 21 21 18
   Looks like we added this dependency before creating the code block. Removing prismjs makes sense because syntax highlighting is the responsibility of a block, not the embedding application. Not removing it means we have to use babel instead of swc, which is slower.

### 📜 Does this require a change to the docs?

No

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1201707629991380/1201839331852112/f) (_internal_)

## 🛡 What tests cover this?

- CI (linting, Playright, unit)
- Local manual tests
- Cross-check on the Þ side:
  - https://github.com/blockprotocol/blockprotocol/pull/224
  - https://blockprotocol-2guzghk1k-hashintel.vercel.app/hub

## ❓ How to test this?

1.  Checkout the branch and install deps
1.  Run the app
1.  Make sure all images are loaded as before
1.  Ensure that syntax in code blocks highlighting works as before

### CODE REVIEW

File: packages/blocks/code/package.json [hunk 1]
```diff
@@ -51,7 +51,7 @@
     "eslint-config-prettier": "8.3.0",
     "eslint-plugin-react": "7.28.0",
     "eslint-plugin-react-hooks": "4.3.0",
-    "html-webpack-plugin": "4.5.1",
+    "html-webpack-plugin-v4": "npm:html-webpack-plugin@4.5.1",
```
Comment (kachkaev) [inline packages/blocks/code/package.json:54]: See PR description

File: packages/hash/shared/src/ProsemirrorSchemaManager.ts [hunk 2]
```diff
@@ -252,7 +252,7 @@ export class ProsemirrorSchemaManager {
     /**
      * @todo we should have some concurrency here
      */
-    for (const componentId of Array.from(componentIds.values())) {
+    for (const componentId of componentIds) {
```
Comment (kachkaev) [inline packages/hash/shared/src/ProsemirrorSchemaManager.ts:255]: Related: https://github.com/hashintel/dev/pull/442#pullrequestreview-847978355 (internal link).

We no longer have `target: es5` in Next, which was the reason for using `Array.from()` in the first place. I’m pretty sure that this code will be transpiled in Next and work in Node 14+. If I’m wrong, I will submit a reverse diff. Adding this change here to make sure we don’t forget about it later.

File: packages/hash/frontend/next-env.d.ts [hunk 3]
```diff
@@ -1,5 +1,4 @@
 /// <reference types="next" />
-/// <reference types="next/types/global" />
```
Comment (kachkaev) [inline packages/hash/frontend/next-env.d.ts:2]: Removed by Next

File: .env [hunk 4]
```diff
@@ -25,6 +25,8 @@ HASH_REDIS_PORT=6379
 HASH_SEARCH_LOADER_PORT=3838
 HASH_SEARCH_QUEUE_NAME=search
 
+LOG_LEVEL=info
+
```
Comment (kachkaev) [inline .env:29]: @benwerner01 I moved `LOG_LEVEL` a bit further up. I did not notice it was under _Aliases for dockerized external services_ when reviewing #322 


Screenshot 2022-02-17 at 17 02 14

Comment (benwerner01) [reply .env:29]: My bad, thank you 🙏

File: packages/hash/frontend/tsconfig.json [hunk 5]
```diff
@@ -1,12 +1,12 @@
 {
   "extends": "../../../tsconfig.base.json",
   "compilerOptions": {
+    "incremental": true,
     "isolatedModules": true,
     "jsx": "preserve",
     "lib": ["dom", "dom.iterable", "esnext"],
-    "target": "es5",
```
Comment (nathggns) [inline packages/hash/frontend/tsconfig.json:7]: Discussing this on Slack, because I'm not sure why this is here.

**Review by nathggns — CHANGES_REQUESTED** [review]

File: packages/hash/shared/src/ProsemirrorSchemaManager.ts (hunk 2 above)
Comment (thehabbos007) [reply packages/hash/shared/src/ProsemirrorSchemaManager.ts:255]: Was this related to `downlevelIteration` not being allowed?

File: packages/hash/frontend/next.config.js [hunk 6]
```diff
@@ -31,59 +30,51 @@ process.env.NEXT_PUBLIC_HASH_OPENSEARCH_ENABLED =
  */
```
Comment (thehabbos007) [inline packages/hash/frontend/next.config.js:30]: > @todo try using next-compose-plugins when upgrading next to 11 and/or to webpack 5
>    was not building with compose-plugins on next 10 w/ webpack 4.

This todo is present above the selected line. Is this relevant with the below changes? If it is, is it relevant for us to capture in a task?

**Review by thehabbos007 — COMMENTED** [review]
I'll leave a comment as @nathggns is requesting changes. I just have a question below. Otherwise looks good. Thank you Alex! Hopefulyl we can get the blocks to webpack 5 after this as well :)

File: packages/hash/frontend/tsconfig.json (hunk 5 above)
Comment (colelawrence) [reply packages/hash/frontend/tsconfig.json:7]: Internal [slack thread](https://hashintel.slack.com/archives/C022217GAHF/p1645186873448139) for reference.

Comment (kachkaev) [reply packages/hash/frontend/tsconfig.json:7]: Reverted in https://github.com/hashintel/hash/pull/320/commits/796c295d06c9917ce0135d1269c11c9bff0cd906

Comment (nathggns) [reply packages/hash/frontend/tsconfig.json:7]: Thanks @kachkaev

File: packages/hash/frontend/next.config.js (hunk 6 above)
Comment (kachkaev) [reply packages/hash/frontend/next.config.js:30]: I was going to add this package first, but then decided not to. We got one less plugin, so fewer things to compose. Maybe we can introduce [next-compose-plugins](https://www.npmjs.com/package/next-compose-plugins) when we need to increase the number of plugins next time. The need for this gets smaller as Next supports more and more features out of box.

**Pull request merged**